linkerd/app/src/env.rs

```diff
@@ -1007,12 +1018,23 @@ fn parse_networks(list: &str) -> Result<HashSet<IpNet>, ParseError> {
 
 fn parse_default_policy(
     s: &str,
+    cluster_nets: HashSet<IpNet>,
     detect_timeout: Duration,
 ) -> Result<policy::DefaultPolicy, ParseError> {
     match s {
         "deny" => Ok(policy::DefaultPolicy::Deny),
         "all-authenticated" => Ok(policy::defaults::all_authenticated(detect_timeout).into()),
         "all-unauthenticated" => Ok(policy::defaults::all_unauthenticated(detect_timeout).into()),
+
+        // If cluster networks are configured, support cluster-scoped default policies.
+        name if cluster_nets.is_empty() => Err(ParseError::InvalidPortPolicy(name.to_string())),
```

Comment:
hmm, should there be a separate error case for this? this way, we will log an error that kind of implies "cluster-authenticated" isn't a valid default policy name, but it _is_ a valid policy, it's just not valid in the case that there's no cluster networks.

possibly improving the error message here is not that important, though, since users aren't setting these env vars manually?

Reply:
> possibly improving the error message here is not that important, though, since users aren't setting these env vars manually?

Correct. Seeing this is basically a programing error: the injector isn't doing the right thing. There's nothing a user could change in their config other than to change/unset the default policy on the pod.